store/SPEC_STANDARD.md

```diff
@@ -0,0 +1,110 @@
+- *Definitions* - A list of new terms or concepts utilised in the document or required to understand it.
+
+### System model and properties
+
+This section should include an assumptions sub-section if any, and the mandatory properties sub-section. Note that both sub-sections are tightly coupled: how to enforce a property will depend directly on the assumptions made.
+
+- *Assumptions* - A list of any assumptions made by the feature designer. This may for instance include how it is assumed that the feature will be used by upper layers in the stack.
+- *Properties* - The content of this sub-section depends on the type of specification. For implementations, the sub-section should include a list of properties that the implementation guarantees and list which properties it does not. The latter is as important as the former. In the case of a store interface, this sub-section should include a list of the desired properties that an implementation of the interface should guarantee. Ideally, along with each of the properties should go a text arguing why the property is required. Examples of properties are whether the feature is or must be thread-safe, atomicity-related properties or behaviour under failures.
+
+### Technical specification
+
+This is the main section of the document, and should contain protocol documentation, design rationale,
+required references, and technical details where appropriate. 
+Apart from the API sub-section which is required, the section may have any or all of the following sub-sections, as appropriate to the particular specification.
+
+- *API* - A detailed description of the features's API including which interfaces implements (for implementations) or extends (for interfaces).
+- *Technical Specification* - All technical details including syntax, diagrams, semantics, protocols, data structures, algorithms, and pseudocode as appropriate. The technical specification should be detailed enough such that separate correct implementations of the specification without knowledge of each other are compatible.
+- *Backwards Compatibility* - A discussion of compatibility (or lack thereof) with previous feature or protocol versions.
+- *Known Issues* - A list of known issues. This sub-section is specially important for specifications of already in-use features.
+- *Example Implementation* - A concrete example implementation or description of an expected implementation to serve as the primary reference for implementers. This sub-section is only expected to be included in newly proposed interfaces.
+
+### History
+
+A store specification should include a history section, listing any inspiring documents and a plaintext log of significant changes.
+
+See an example history section [below](#history-1).
+
+### Copyright
+
+A store specification should include a copyright section waiving rights via [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0).
```

Comment:
is this needed because the repo has an apache license?

Reply:
Problem is that what if the document is shared out of this repo? I think keeping the copyright as is ensures someone can't steal it and pretend they didn't know it was copyright of Cosmos Ecosystem, claiming they just found the markdown file online in a random source such as a Github Gist or on some website.